### PR TITLE
Pytorch parallelization

### LINKED ISSUE #5870: [Concept Entry] PyTorch: Parallelizing Models
### Reason/inspiration (optional)

We would like a new entry on `Parallelizing Models` concept under PyTorch. The entry should go in a new file under `docs/content/pytorch/concepts/parallelizing-models/parallelizing-models.md`.       
  
The entry should include:      
  
- An introduction to the concept      
- A `Syntax` section that provides the syntax for the concept (if applicable)      
- An `Example` section that provides an example demonstrating the concept in use (if applicable)     

 
Please refer to the [concept entry template](https://github.com/Codecademy/docs/blob/main/documentation/concept-entry-template.md), [content standards](https://github.com/Codecademy/docs/blob/main/documentation/content-standards.md), and [markdown style guide](https://curriculum-documentation.codecademy.com/content-guidelines/markdown-style-guide/) when working on the PR for this issue.|

### Entry Type (select all that apply)

New Concept Entry for Existing Topic

### Code of Conduct

- [X] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [X] Labels added
- [ ] Issue is assigned

### PR DESCRIPTION
**Pull request by parkersarahl**
### Description

A concept entry was created for parallelizing models, under PyTorch. 

### Issue Solved

Closes #5870 

### Type of Change

- Adding a new entry

### Checklist

- [X ] All writings are my own.
- [ X] My entry follows the Codecademy Docs style guide.
- [ X] My changes generate no new warnings.
- [ X] I have performed a self-review of my own writing and code.
- [ X] I have checked my entry and corrected any misspellings.
- [ X] I have made corresponding changes to the documentation if needed.
- [ X] I have confirmed my changes are not being pushed from my forked `main` branch.
- [ X] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [ X] I have linked any issues that are relevant to this PR in the `Issues Solved` section.

### CODE REVIEW

File: content/pytorch/concepts/parallelizing-models/parallelizing-models.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  - 'Machine Learning'
+  - 'Data Science'
+Tags: 
+  - 'Algorithms'
+  - 'PyTorch'
+  - 'Machine Learning'
+CatalogContent: 
+  - 'intro-to-py-torch-and-neural-networks'
+  - 'paths/build-a-machine-learning-model'
+---
```
Comment (mamtawardhani) [inline content/pytorch/concepts/parallelizing-models/parallelizing-models.md:15]:
```suggestion
---
Title: 'parallelizing-models' 
Description: 'PyTorch parallelizing models distribute computations across multiple processors or GPUs, improving performance and enabling faster training of large models.' 
Subjects:
  - 'Computer Science'
  - 'Machine Learning'
  - 'Data Science'
Tags: 
  - 'Algorithms'
  - 'PyTorch'
  - 'Machine Learning'
CatalogContent: 
  - 'intro-to-py-torch-and-neural-networks'
  - 'paths/build-a-machine-learning-model'
---
```

File: content/pytorch/concepts/parallelizing-models/parallelizing-models.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+**Model parallelization** trains deep learning models that require more memory than what is available on a single graphic processing unit (GPU). It is used within the pyTorch library. The model is separated into parts (i.e. layers or modules) and assigned to different GPUs. The GPUs perform the computations simultaneously, allowing for faster processing of large models. The GPUs communicate with each other and share the data to ensure the data output from one GPU is used in another GPU as needed.
```
Comment (mamtawardhani) [inline content/pytorch/concepts/parallelizing-models/parallelizing-models.md:17]:
```suggestion
**Model parallelization** in PyTorch allows the training of deep learning models that require more memory than a single GPU can provide. The model is divided into different parts (e.g., layers or modules), with each part assigned to a separate GPU. These GPUs perform computations simultaneously, speeding up the processing of large models. They communicate with each other and share data to ensure that the output from one GPU can be used by another when necessary.
```

File: content/pytorch/concepts/parallelizing-models/parallelizing-models.md [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+---
+
+**Model parallelization** trains deep learning models that require more memory than what is available on a single graphic processing unit (GPU). It is used within the pyTorch library. The model is separated into parts (i.e. layers or modules) and assigned to different GPUs. The GPUs perform the computations simultaneously, allowing for faster processing of large models. The GPUs communicate with each other and share the data to ensure the data output from one GPU is used in another GPU as needed. 
+
+## Setting up the environment
+```pseudo 
+import torch
+import torch.nn as nn
+```
+
```
Comment (mamtawardhani) [inline content/pytorch/concepts/parallelizing-models/parallelizing-models.md:24]: This part can be removed

File: content/pytorch/concepts/parallelizing-models/parallelizing-models.md [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+## Syntax
+To utilize model parallelization the model should be wrapped using the following syntax:
+
+```shell
+class ModelParallel(nn.Module)
```
Comment (mamtawardhani) [inline content/pytorch/concepts/parallelizing-models/parallelizing-models.md:29]:
```suggestion
class ModelParallel(nn.Module):
    # Model definition goes here
```

File: content/pytorch/concepts/parallelizing-models/parallelizing-models.md [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+        x = self.layer1(x)
+        x = x.to('cuda:1')  # Output of first layer to second GPU
+        x = self.layer2(x)
+        return x
+
+model = ModelParallel()
+x = torch.randn(64, 1000)
+output = model(x)
+```
+## Output
```
Comment (mamtawardhani) [inline content/pytorch/concepts/parallelizing-models/parallelizing-models.md:54]:
```suggestion
```

File: content/pytorch/concepts/parallelizing-models/parallelizing-models.md [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+    def forward(self, x):
+        x = x.to('cuda:0')  # Input to first GPU
+        x = self.layer1(x)
+        x = x.to('cuda:1')  # Output of first layer to second GPU
+        x = self.layer2(x)
+        return x
+
+model = ModelParallel()
+x = torch.randn(64, 1000)
+output = model(x)
```
Comment (mamtawardhani) [inline content/pytorch/concepts/parallelizing-models/parallelizing-models.md:52]:
```suggestion
import torch.nn as nn

# Define a model split across two GPUs
class ModelParallel(nn.Module):
    def __init__(self):
        super(ModelParallel, self).__init__()
        self.layer1 = nn.Linear(1000, 500).to('cuda:0')  # First GPU
        self.layer2 = nn.Linear(500, 100).to('cuda:1')   # Second GPU

    def forward(self, x):
        x = x.to('cuda:0')  # Input to first GPU
        x = self.layer1(x)
        x = x.to('cuda:1')  # Output of first layer to second GPU
        x = self.layer2(x)
        return x

model = ModelParallel()
x = torch.randn(64, 1000)
output = model(x)
```

**Review by mamtawardhani — CHANGES_REQUESTED** [review]
Hey @parkersarahl, thank you for contributing to Codecademy Docs, the entry is nicely written! 😄 

I've suggested a few changes, could you please review and modify those at your earliest convenience? Thank you! 😃

**Comment by parkersarahl** [comment]
@mamtawardhani I made the changes as suggested for your review. Let me know if anything else needs to be updated. Thank you 😃

**Review by mamtawardhani — APPROVED** [review]
Thank you for contributing to Codecademy Docs @parkersarahl  😄 

The entry looks good for a second round of review! 🚀

File: content/pytorch/concepts/parallelizing-models/parallelizing-models.md [hunk 7]
```diff
@@ -0,0 +1,61 @@
+---
+Title: 'Parallelizing Models'
+Description: 'PyTorch parallelizing models distribute computations across multiple processors or GPUs, improving performance and enabling faster training of large models.'
+Subjects:
+  - 'Computer Science'
+  - 'Machine Learning'
+  - 'Data Science'
```
Comment (Radhika-okhade) [inline content/pytorch/concepts/parallelizing-models/parallelizing-models.md:7]:
```suggestion
  - 'Computer Science'
  - 'Data Science'
  - 'Machine Learning'
```

File: content/pytorch/concepts/parallelizing-models/parallelizing-models.md [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+Tags:
+  - 'Algorithms'
+  - 'PyTorch'
+  - 'Machine Learning'
```
Comment (Radhika-okhade) [inline content/pytorch/concepts/parallelizing-models/parallelizing-models.md:11]:
```suggestion
  - 'Algorithms'
  - 'Machine Learning'
  - 'PyTorch'
```

**Review by Radhika-okhade — APPROVED** [review]

**Comment by Radhika-okhade** [comment]
Thank you for contributing to Codecademy docs @parkersarahl. The entry looks good to be merged.
